src/nvim/memline.c

```diff
@@ -131,7 +131,8 @@ struct data_block {
   unsigned db_free;             // free space available
   unsigned db_txt_start;        // byte where text starts
   unsigned db_txt_end;          // byte just after data block
-  linenr_T db_line_count;       // number of lines in this block
+  // linenr_T db_line_count;
+  long db_line_count;           // number of lines in this block
```

Comment:
I guess this needs to have the same size as `b0_magic_long`. Maybe changing some other variables to `linenr_T` can work too, but I haven't figured out how.